base/commands/migration/start_stages.go

```diff
@@ -0,0 +1,219 @@
+	}
+	st.updateMsgChan <- u
+}
+
+func (st *StartStages) startStage(ctx context.Context, ec plug.ExecContext) func(stage.Statuser) error {
+	return func(stage.Statuser) error {
+		cb, err := makeConfigBundle(st.configDir, st.migrationID)
+		if err != nil {
+			return err
+		}
+		if err = st.startQueue.Put(ctx, cb); err != nil {
+			return err
+		}
+		ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+		defer cancel()
+		if isTerminal, err := st.handleUpdateMessage(ctx, ec, <-st.updateMsgChan); isTerminal {
+			return err
+		}
+		return nil
+	}
+}
+
+func makeConfigBundle(configDir, migrationID string) (serialization.JSON, error) {
+	var cb ConfigBundle
+	cb.MigrationID = migrationID
+	if err := cb.Walk(configDir); err != nil {
+		return nil, err
+	}
+	b, err := json.Marshal(cb)
+	if err != nil {
+		return nil, err
+	}
+	return b, nil
+}
+
+func (st *StartStages) migrateStage(ctx context.Context, ec plug.ExecContext) func(statuser stage.Statuser) error {
+	return func(stage.Statuser) error {
+		for {
+			select {
+			case msg := <-st.updateMsgChan:
+				if isTerminal, err := st.handleUpdateMessage(ctx, ec, msg); isTerminal {
+					return err
+				}
+			case <-ctx.Done():
+				if err := ctx.Err(); err != nil {
+					if errors.Is(err, context.DeadlineExceeded) {
+						return timeoutErr
+					}
+					return fmt.Errorf("migration failed: %w", err)
+				}
+			}
+		}
+	}
+}
+
+func (st *StartStages) handleUpdateMessage(ctx context.Context, ec plug.ExecContext, msg UpdateMessage) (bool, error) {
+	ec.PrintlnUnnecessary(msg.Message)
+	if slices.Contains([]Status{StatusComplete, StatusFailed, StatusCanceled}, msg.Status) {
+		ms, err := readMigrationStatus(ctx, st.statusMap)
+		if err != nil {
+			return true, fmt.Errorf("reading status: %w", err)
+		}
+		ec.PrintlnUnnecessary(ms.Report)
+		name := fmt.Sprintf("migration_report_%s", st.migrationID)
+		if err = saveReportToFile(name, ms.Report); err != nil {
+			return true, fmt.Errorf("writing report to file: %w", err)
+		}
+		if err = st.saveDebugLogs(ctx, st.ci.OrderedMembers()); err != nil {
+			return true, fmt.Errorf("writing debug logs to file: %w", err)
+		}
+		ec.PrintlnUnnecessary(fmt.Sprintf("migration report saved to file: %s", name))
+		for _, l := range ms.Logs {
+			ec.Logger().Info(l)
+		}
+		switch ms.Status {
+		case StatusComplete:
+			return true, nil
+		case StatusCanceled:
+			return true, clcerrors.ErrUserCancelled
+		case StatusFailed:
+			return true, fmt.Errorf("migration failed")
+		}
+	}
+	return false, nil
+}
+
+func saveReportToFile(fileName, report string) error {
+	f, err := os.Create(fmt.Sprintf(fileName))
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	_, err = f.WriteString(report)
+	return err
+}
+
+func (st *StartStages) saveDebugLogs(ctx context.Context, members []cluster.MemberInfo) error {
+	for _, m := range members {
+		f, err := os.Create(fmt.Sprintf("%s%s.log", DebugLogsListPrefix, m.UUID.String()))
```

Comment:
I think we can just use CLC logging for debug logs.
We can prefix the log lines with migration ID and member UUID.